Document Sec-CH-Prefers-Color-Scheme Client Hint header (#22266)
mdn/content: files/en-us/web/css/@media/index.md

- `or`
  - : Equivalent to the `,` operator. Added in Media Queries Level 4.

> ### User Agent Client Hints

Reviewer: IMO we don't treat User Agent as a proper noun anywhere else in MDN right. The only place this should be capitalized is  if we are referring specifically/in-full to the "User Agent Client Hints API", which is a proper name. 

```suggestion
### User agent client hints
```

> Some media queries have corresponding [User Agent client hints](/en-US/docs/Web/HTTP/Client_hints).

Reviewer: ```suggestion
Some media queries have corresponding [user agent client hints](/en-US/docs/Web/HTTP/Client_hints).
```

These are HTTP headers that request content that is pre-optimized for the particular media requirement.

> They include: {{HTTPHeader("Sec-CH-Prefers-Color-Scheme")}}, {{HTTPHeader("Sec-CH-Prefers-Reduced-Motion")}}, and {{HTTPHeader("Save-Data")}}.

Reviewer: `Save-Data` is not a user-agent client hint and it is implied that it is by the first link in this paragraph. It used to be a network client hint, and now it is is "unclear" - it recently moved into the save-data API. Network client hint is still the closest grouping.

Perhaps just swap to another hint that has a media query?

Reviewer 2: I've just removed `Save-Data`, reducing the list down to two, as this paragraph seems to be talking explicitly about user agent client hints that specific mirror media query functionality. And with this, PR, we'll only have those two documented ;-)

Both [`em`](/en-US/docs/Learn/CSS/Building_blocks/Values_and_units#numeric_values) and [`px`](/en-US/docs/Learn/CSS/Building_blocks/Values_and_units#numeric_values) are valid units, but [`em`](/en-US/docs/Learn/CSS/Building_blocks/Values_and_units#numeric_values) works better if the user changes the browser text size.

> Also consider media queries or [HTTP user agent client hints](/en-US/docs/Web/HTTP/Client_hints#user-agent_client_hints) to improve the user's experience.

Reviewer: FYI, I added this.

Reviewer 2: Looks good

For example, the media query [`prefers-reduced-motion`](/en-US/docs/Web/CSS/@media/prefers-reduced-motion) or the eqivalent HTTP header {{HTTPHeader("Sec-CH-Prefers-Reduced-Motion")}}) can be used to minimize the amount of animation or motion used based on user preferences.

## Security

Because media queries provide insights into the capabilities—and by extension, the features and design—of the device the user is working with, there is the potential that they could be abused to construct a "fingerprint" which identifies the device, or at least categorizes it to some degree of detail that may be undesirable to users.

Because of this potential, a browser may opt to fudge the returned values in some manner in order to prevent them from being used to precisely identify a computer. A browser might also offer additional measures in this area; for example, if Firefox's "Resist Fingerprinting" setting is enabled, many media queries report default values rather than values representing the actual device state.

> ## User Agent Client Hints

Reviewer: Actually not entirely sure about this whole section - its not like security or accessibility, which are considerations about THIS API.

Maybe this could be a level 3 heading under description? Not sure. It is worth mentioning.

Reviewer 2: Moved in my latest commit, as suggested

Because of this potential, a browser may opt to fudge the returned values in some manner in order to prevent them from being used to precisely identify a computer. A browser might also offer additional measures in this area; for example, if Firefox's "Resist Fingerprinting" setting is enabled, many media queries report default values rather than values representing the actual device state.

## User Agent Client Hints

> Some media queries have corresponding [User Agent client hint](/en-US/docs/Web/HTTP/Client_hints#user-agent_client_hints) which represent the values of discrete headers as HTTP response headers. These headers include: {{HTTPHeader("Sec-CH-Prefers-Color-Scheme")}}, {{HTTPHeader("Sec-CH-Prefers-Reduced-Motion")}}, and {{HTTPHeader("Sec-CH-Save-Data")}}.

Reviewer: ```suggestion
Some media queries have corresponding [User Agent client hints](/en-US/docs/Web/HTTP/Client_hints#user-agent_client_hints), which are HTTP headers that request content that is already customized for the particular media requirement. These headers include: {{HTTPHeader("Sec-CH-Prefers-Color-Scheme")}}, {{HTTPHeader("Sec-CH-Prefers-Reduced-Motion")}}, and {{HTTPHeader("Sec-CH-Save-Data")}}.
```

Because of this potential, a browser may opt to fudge the returned values in some manner in order to prevent them from being used to precisely identify a computer. A browser might also offer additional measures in this area; for example, if Firefox's "Resist Fingerprinting" setting is enabled, many media queries report default values rather than values representing the actual device state.

## User Agent Client Hints

> Some media queries have corresponding [User Agent client hints](/en-US/docs/Web/HTTP/Client_hints#user-agent_client_hints), which are HTTP headers that request content that is already customized for the particular media requirement. These headers include: {{HTTPHeader("Sec-CH-Prefers-Color-Scheme")}}, {{HTTPHeader("Sec-CH-Prefers-Reduced-Motion")}}, and {{HTTPHeader("Save-Data")}}.

Reviewer 2: Bit of a nit, but you could remove "that is already" from this sentence to tighten it up a bit.

Reviewer: ```suggestion
Some media queries have corresponding [User Agent client hints](/en-US/docs/Web/HTTP/Client_hints#user-agent_client_hints), which are HTTP headers that request content that are customized for the particular media requirement. These headers include: {{HTTPHeader("Sec-CH-Prefers-Color-Scheme")}}, {{HTTPHeader("Sec-CH-Prefers-Reduced-Motion")}}, and {{HTTPHeader("Save-Data")}}.
```
